File: gitok/lib/core/utils/logger.dart
import 'dart:developer' as developer;

/// 日志工具类
class Logger {
  void info(String message) {
    developer.log(message, name: 'INFO');
  }

  void error(String message, [dynamic error]) {
    developer.log('$message${error != null ? '\n$error' : ''}', name: 'ERROR');
  }
}

File: packages/plugin-ide/src/utils/logger.ts
/**
 * 日志工具
 */
export class Logger {
  private tag: string;

  constructor(tag: string) {
    this.tag = tag;
  }

  info(message: string, ...args: any[]) {
    console.log(`[${this.tag}] ${message}`, ...args);
  }

  error(message: string, ...args: any[]) {
    console.error(`[${this.tag}] ${message}`, ...args);
  }

  debug(message: string, ...args: any[]) {
    console.log(`[${this.tag}:调试] ${message}`, ...args);
  }
}
